src/Files.App/ViewModels/SettingsViewModels/AdvancedViewModel.cs

```diff
@@ -299,5 +317,72 @@ private FileOpenPicker InitializeWithWindow(FileOpenPicker obj)
 
 			return obj;
 		}
+
+		private void DoAddNewTag()
+		{
+			NewTag.Reset();
+			IsCreatingNewTag = true;
+		}
+
+		private void DoSaveNewTag()
+		{
+			IsCreatingNewTag = false;
+
+			fileTagsSettingsService.CreateNewTag(NewTag.Name, NewTag.Color);
+			Tags.Clear();
+			fileTagsSettingsService.FileTagList?.ForEach(tag => Tags.Add(new ListedTagViewModel(tag)));
+		}
+
+		private void DoCancelNewTag()
+		{
+			IsCreatingNewTag = false;
+		}
+
+		public void EditExistingTag(ListedTagViewModel item, string newName, string color)
+		{
+			fileTagsSettingsService.EditTag(item.Tag.Uid, newName, color);
+			Tags.Clear();
+			fileTagsSettingsService.FileTagList?.ForEach(tag => Tags.Add(new ListedTagViewModel(tag)));
+		}
+
+		public void DeleteExistingTag(ListedTagViewModel item)
+		{
+			Tags.Remove(item);
+			fileTagsSettingsService.DeleteTag(item.Tag.Uid);
+		}
+	}
+
+	public class NewTagViewModel : ObservableObject
+	{
+		private string name = string.Empty;
+		public string Name
+		{
+			get => name;
+			set
+			{
+				if (SetProperty(ref name, value))
+					OnPropertyChanged(nameof(IsNameValid));
+			}
+		}
+
+		private string color = "#FFFFFFFF";
+		public string Color
+		{
+			get => color;
+			set => SetProperty(ref color, value);
+		}
+
+		private bool isNameValid;
+		public bool IsNameValid
+		{
+			get => isNameValid;
+			set => SetProperty(ref isNameValid, value);
+		}
+
+		public void Reset()
+		{
+			Name = string.Empty;
+			Color = ColorHelpers.RandomColor();
+		}
```

Comment:
Worth moving into its own file to not clutter the class.

Reply:
Where should the file go?

Reply:
Maybe in `DataModels` or `Backed.ViewModels.FileTags`?

Reply:
I would have suggested still in `ViewModels`, but in an entire new class. But after looking at the code from within the Visual Studio, I see the class is only used insside the `AvancedViewModel`, so if we're sure it will never be used elsewhere, it may stay there.
May I ask what is the use of the `Backed` namespace?

Reply:
I forgot an `N` it should have been `BackeNd.ViewModels.FileTags`, which is where other Tags view models are

Reply:
In that case, that might be a good idea!